src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -233,13 +229,25 @@ private bool DetermineExtensionDataProperty(Dictionary<string, JsonPropertyInfo>
             if (jsonPropertyInfo != null)
             {
                 Type declaredPropertyType = jsonPropertyInfo.DeclaredPropertyType;
-                if (!typeof(IDictionary<string, JsonElement>).IsAssignableFrom(declaredPropertyType) &&
-                    !typeof(IDictionary<string, object>).IsAssignableFrom(declaredPropertyType))
+                if (typeof(Dictionary<string, object>).IsAssignableFrom(declaredPropertyType) ||
```

Comment:
Yes we need them in case there is a custom collection that implements `IDictionary<>` but doesn't derive from `Dictionary<>`

Reply:
> Yes we need them in case there is a custom collection that implements `IDictionary<>` but doesn't derive from `Dictionary<>`

If a collection implements `IDictionary<>` then the `typeof(IDictionary<string, object>).IsAssignableFrom(declaredPropertyType)` check would be true, no? Why does it matter whether it derived from `Dictionary<>` or not?

Reply:
Yep will change. Remnant from a previous implementation in this effort there were 4 special converters for all 4 combinations (instead of trying to use the built-in dictionary converter for extension data) and it needed to map to each one separately.